kanidmd_web_ui/src/components/admin_accounts.rs

```diff
@@ -0,0 +1,258 @@
+use crate::components::adminmenu::{Entity, GetError};
+use crate::components::alpha_warning_banner;
+use crate::constants::{CSS_CELL, CSS_PAGE_HEADER, CSS_TABLE};
+use crate::models;
+use crate::utils::{do_alert_error, init_request};
+use gloo::console;
+use std::collections::BTreeMap;
+// use yew::prelude::*;
+use yew::{html, Component, Context, Html, Properties};
+
+impl From<GetError> for AdminListAccountsMsg {
+    fn from(ge: GetError) -> Self {
+        AdminListAccountsMsg::Failed {
+            emsg: ge.err,
+            kopid: None,
+        }
+    }
+}
+
+pub struct AdminListAccounts {
+    state: ViewState,
+}
+
+// callback messaging for this confused pile of crab-bait
+pub enum AdminListAccountsMsg {
+    /// When the server responds and we need to update the page
+    Responded {
+        response: BTreeMap<String, Entity>,
+    },
+    Failed {
+        emsg: String,
+        kopid: Option<String>,
+    },
+}
+
+enum ViewState {
+    /// waiting for the page to load
+    Loading,
+    /// server has responded
+    Responded { response: BTreeMap<String, Entity> },
+    /// failed to pull the details
+    #[allow(dead_code)]
+    Failed {
+        // TODO: use this
+        emsg: String,
+        kopid: Option<String>,
+    },
+    #[allow(dead_code)]
+    /// Not authorized to pull the details
+    NotAuthorized {}, // TODO: use this
+}
+
+#[derive(PartialEq, Properties, Eq)]
+pub struct AdminListAccountsProps {
+    // for filtering and pagination
+    // #[allow(dead_code)]
+    // search: Option<String>,
+    // #[allow(dead_code)]
+    // page: Option<u32>,
+}
+
+/// Pulls all accounts (service or person-class) from the backend and returns a HashMap
+/// with the "name" field being the keys, for easy human-facing sortability.
+pub async fn get_accounts(token: &str) -> Result<AdminListAccountsMsg, GetError> {
```

Comment:
I think the more we put into  the webadmin like this, the more important it'll be we start to test it, because else there is a lot of risk of breakage especially around api changes.

Reply:
Yep, that's going to be... fun :)

I've got some testing automation to build up (create 10K random accounts, to test how the API/UI blows up, etc) which is going to come after this PR's done, then I'll probably throw myself on the browser-testing grenade.

Reply:
Well I actually don't think that'll be too bad, I think you can probably re-use some of the ideas from https://github.com/kanidm/kanidm/blob/master/kanidm_unix_int/tests/cache_layer_test.rs and use that to spin up a server and then test the "apis" here without actually needing a browser tester like selenium at all. That will cover a big chunk of the "wht can go wrong" at least. And I think yew can be tested from cargo test like this, which can cover the rest of the interactions. So I think it's probably not too bad. but we might need to consider breaking ` async fn setup_test` into it's own crate because it's a duplicate of the setup for proto_test in the server itself.

Reply:
Yeah, that's fair - I guess that's one of the advantages of using rust/WASM, it *should* be largely testable outside a browser of some kind 😃